### PR TITLE
feat(log): log errors in a more consistent fashion

### PR DESCRIPTION
**Pull request by rs-amp**
This PR changes the reporting of errors and warnings into the log to be more consistent, and machine readable.

When an error occurs, there is now an `ERROR` or `WARNING` action inserted into the logs.

When any error or warning occurs, the overall error level of the log is increased to match the error. This is printed at the end of the file as `SUCCESS` if there are no warnings or errors, `FAILURE` if there are errors, or the highest error level otherwise.

The errors logs themselves have been made more consistent - logs with `Error` instances will now print them on the next line all of the time, rather than inconsistently as part of the string template.

This shouldn't be a breaking change. The final return code is on the last line of the log, which was previously empty. (the log ended with a new line). If the `ArchiveLog` attempts to parse a log without an exit code, it should default to `SUCCESS`.

### CODE REVIEW

**Comment by rs-amp** [comment]
Note: The archive/unarchive commands only begin logging after they have found the content to archive, so any errors during that time will cause the command to exit without producing a log. Errors for these commands will only be logged when the error happens during the archive/unarchive process, otherwise no log file will be created.

This PR doesn't change this behaviour, but it might be changed in future.

**Comment by easen-amp** [comment]
Having `log.addError(LogErrorLevel.WARNING, ...)` and `log.addError(LogErrorLevel.ERROR,...)` doesn't look right to me, normally a logger would have a series of methods for each level that it supports (`log.debug()`, `log.info()`, `log.warn()`, `log.error()`) and the 1st argument passed into it would be a `string` and the next/remaining arguments would be contextual data (additional data that needs to be logged), e.g. 
```typescript
try {
  myMethod();
} catch (err) {
  log.error('Failed to do X', {error: err});
} ```

See https://www.npmjs.com/package/js-logger for a more concrete example.

Can this be updated to follow this pattern?

**Review by easen-amp — CHANGES_REQUESTED** [review]
Having `log.addError(LogErrorLevel.WARNING, ...)` and `log.addError(LogErrorLevel.ERROR,...)` doesn't look right to me, normally a logger would have a series of methods for each level that it supports (`log.debug()`, `log.info()`, `log.warn()`, `log.error()`) and the 1st argument passed into it would be a `string` and the next/remaining arguments would be contextual data (additional data that needs to be logged), e.g. 
```typescript
try {
  myMethod();
} catch (err) {
  log.error('Failed to do X', {error: err});
} 
```

See https://www.npmjs.com/package/js-logger for a more concrete example.

Can this be updated to follow this pattern?

**Comment by rs-amp** [comment]
I've updated the warn and error logs to use special methods, and the `addError` function has been privated as a result. The enum is still used for severity and parsing. Is this alright?

File: src/commands/content-item/archive.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Arguments, Argv } from 'yargs';
 import { ConfigurationParameters } from '../configure';
 import dynamicContentClientFactory from '../../services/dynamic-content-client-factory';
-import { ArchiveLog } from '../../common/archive/archive-log';
+import { LogErrorLevel, ArchiveLog } from '../../common/archive/archive-log';
```
Comment (easen-amp) [inline src/commands/content-item/archive.ts:4]: checkstyle warning

File: src/commands/content-item/archive.ts [hunk 2]
```diff
@@ -265,13 +265,9 @@ export const processItems = async ({
       log.addComment(e.toString());
 
       if (ignoreError) {
-        console.log(
-          `Failed to archive ${contentItems[i].label} (${contentItems[i].id}), continuing. Error: \n${e.toString()}`
-        );
+        log.warn(`Failed to archive ${contentItems[i].label} (${contentItems[i].id}), continuing.`, e);
```
Comment (easen-amp) [inline src/commands/content-item/archive.ts:268]: much better 👍

File: src/commands/content-item/import.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ import { asyncQuestion } from '../../common/archive/archive-helpers';
 import { AmplienceSchemaValidator, defaultSchemaLookup } from '../../common/content-item/amplience-schema-validator';
 import { getDefaultLogPath } from '../../common/log-helpers';
 import { PublishQueue } from '../../common/import/publish-queue';
+import { LogErrorLevel } from '../../common/archive/archive-log';
```
Comment (easen-amp) [inline src/commands/content-item/import.ts:34]: check style warnings

**Review by PikinerLiliya — APPROVED** [review]
Good job

**Review by easen-amp — CHANGES_REQUESTED** [review]
Looks better, just some checkstyle warnings from the not used `LogErrorLevel` imports

File: src/commands/content-item/import.ts (hunk 3 above)
Comment (rs-amp) [reply src/commands/content-item/import.ts]: Oops, of course I forgot to remove these 🥴

**Review by easen-amp — APPROVED** [review]

**Pull request merged**